tests/ui_tests/test_ui_pixel_options_widget.py

```diff
@@ -439,3 +586,256 @@ def test_UI_GIVEN_no_pixels_button_is_pressed_WHEN_entering_pixel_data_THEN_call
+        == pixel_grid.count_direction
+    )
+
+
+@pytest.mark.parametrize("corner", INITIAL_COUNT_CORNER.values())
+def test_GIVEN_detector_numbers_WHEN_calling_get_detector_number_information_THEN_expected_start_counting_are_returned(
+    pixel_options, pixel_grid, corner, off_geometry_file, off_component_with_pixel_grid
+):
+    pixel_grid.initial_count_corner = corner
+    replace_pixel_grid_in_off_component(
+        off_component_with_pixel_grid, pixel_grid, off_geometry_file
+    )
+    pixel_options.fill_existing_entries(off_component_with_pixel_grid)
+
+    assert (
+        INITIAL_COUNT_CORNER[pixel_options.start_counting_combo_box.currentText()]
+        == corner
+    )
+
+
+def test_GIVEN_component_with_pixel_mapping_WHEN_editing_pixel_data_THEN_correct_number_of_mapping_widgets_are_created(
+    pixel_options, pixel_mapping_with_six_pixels, off_component_with_pixel_mapping
+):
+    pixel_options.fill_existing_entries(off_component_with_pixel_mapping)
+    assert len(pixel_options.pixel_mapping_widgets) == len(
+        pixel_mapping_with_six_pixels.pixel_ids
+    )
+
+
+def test_GIVEN_component_with_pixel_mapping_WHEN_editing_pixel_data_THEN_correct_ids_are_present(
+    pixel_options, pixel_mapping_with_six_pixels, off_component_with_pixel_mapping
+):
+    pixel_options.fill_existing_entries(off_component_with_pixel_mapping)
+    assert widgets_match_pixel_mapping(pixel_mapping_with_six_pixels, pixel_options)
+
+
+def test_GIVEN_component_with_single_id_WHEN_editing_pixel_data_THEN_correct_number_of_mapping_widgets_are_created(
+    pixel_options,
+    pixel_mapping_with_six_pixels,
+    off_component_with_pixel_mapping,
+    off_geometry_file,
+):
+    pixel_mapping_with_six_pixels.pixel_ids = [
+        None for _ in range(CORRECT_CUBE_FACES - 1)
+    ] + [4]
+    replace_pixel_mapping_in_off_component(
+        off_component_with_pixel_mapping,
+        pixel_mapping_with_six_pixels,
+        off_geometry_file,
+    )
+    pixel_options.fill_existing_entries(off_component_with_pixel_mapping)
+
+    assert widgets_match_pixel_mapping(pixel_mapping_with_six_pixels, pixel_options)
+
+
+def test_GIVEN_cylindrical_geometry_WHEN_editing_pixel_mapping_with_single_pixel_THEN_pixel_data_is_recovered(
+    pixel_options,
+    cylindrical_component_with_pixel_mapping,
+    nx_cylindrical_geometry_group,
+    pixel_mapping_with_single_pixel,
+):
+    cylindrical_geometry = cylindrical_component_with_pixel_mapping.set_cylinder_shape(
+        pixel_data=pixel_mapping_with_single_pixel
+    )
+    pixel_options.fill_existing_entries(cylindrical_component_with_pixel_mapping)
+
+    n_cylinders = cylindrical_geometry.cylinders.size / 3
+    assert n_cylinders == len(pixel_options.pixel_mapping_widgets)
+    assert (
+        pixel_options.pixel_mapping_widgets[0].id
+        == pixel_mapping_with_single_pixel.pixel_ids[0]
+    )
+
+
+@pytest.mark.parametrize("old_value", [5, 15])
+@pytest.mark.parametrize("new_value", [5, 15])
```

Comment:
I don't understand what this is testing, are old and new value supposed to be the same?